### PR TITLE
[xlang] Cross language serialization for ActorHandle 

### PR DESCRIPTION
**Pull request by fyrestone**
Problem: Users need to serialize the `ActorHandle` to bytes manually to make it cross-language.

Currently, I have not found a perfect solution for serializing custom types across different languages. So, this implementation has some limitations:
- A cross-language container with ActorHandle elements will be treated as language specific data, it's not cross-language. e.g. `data = [actor_handle]` can't be transferred from Python to Java.
- A special metadata for the `ActorHande` type.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: java/test/src/main/java/io/ray/test/CrossLanguageInvocationTest.java [hunk 1]
```diff
@@ -301,9 +299,8 @@ public static String returnInputString(String s) {
     return l;
   }
 
-  public static byte[] callPythonActorHandle(byte[] value) {
+  public static byte[] callPythonActorHandle(NativePyActorHandle actor) {
```
Comment (kfstorm) [inline java/test/src/main/java/io/ray/test/CrossLanguageInvocationTest.java:302]:
```suggestion
  public static byte[] callPythonActorHandle(PyActorHandle actor) {
```

File: python/ray/ray_constants.py [hunk 2]
```diff
@@ -192,6 +192,10 @@ def to_memory_units(memory_bytes, round_up):
 # A constant used as object metadata to indicate the object is raw bytes.
 OBJECT_METADATA_TYPE_RAW = b"RAW"
 
+# A constant used as object metadata for ActorHandle
```
Comment (kfstorm) [inline python/ray/ray_constants.py:195]:
```suggestion
# A constant used as object metadata to indicate the object is an actor handle.
```

File: python/ray/ray_constants.py [hunk 3]
```diff
@@ -192,6 +192,10 @@ def to_memory_units(memory_bytes, round_up):
...
+# TODO(fyrestone): Serialize the ActorHandle as a custom type of XLANG.
```
Comment (kfstorm) [inline python/ray/ray_constants.py:196]:
```suggestion
# TODO(fyrestone): Serialize the ActorHandle via the custom type feature of XLANG.
```

File: python/ray/serialization.py [hunk 4]
```diff
@@ -354,6 +357,14 @@ def _serialize_to_msgpack(self, value):
             metadata = str(
                 ErrorType.Value("TASK_EXECUTION_EXCEPTION")).encode("ascii")
             value = value.to_bytes()
+        elif isinstance(value, ray.actor.ActorHandle):
+            # TODO(fyresone): ActorHandle should be serialized as a
+            #  custom cross language type.
```
Comment (kfstorm) [inline python/ray/serialization.py:362]:
```suggestion
            # TODO(fyresone): ActorHandle should be serialized via the custom type feature of cross-language.
```

File: python/ray/serialization.py [hunk 5]
```diff
@@ -354,6 +357,14 @@ def _serialize_to_msgpack(self, value):
             metadata = str(
                 ErrorType.Value("TASK_EXECUTION_EXCEPTION")).encode("ascii")
             value = value.to_bytes()
+        elif isinstance(value, ray.actor.ActorHandle):
```
Comment (kfstorm) [inline python/ray/serialization.py:360]: Why not put the logic into `actor_handle_reducer`?

Comment (fyrestone) [reply python/ray/serialization.py:360]: `actor_handle_reducer` is for cloud pickle (e.g. a list of ActorHandle passed from Python to Python), while this logic is for cross language serialization.

Comment (kfstorm) [reply python/ray/serialization.py:360]: I see. But can you try to refactor them to avoid duplicated code?

Comment (fyrestone) [reply python/ray/serialization.py:362]: I think it's not necessary to merge them together:
- Only one line is duplicated: `serialized, actor_handle_id = value._serialization_helper()`
- Current implementation is not the final version, the logic will be changed after we have `custom cross-language type` feature.

File: python/ray/ray_constants.py [hunk 6]
```diff
@@ -192,6 +192,11 @@ def to_memory_units(memory_bytes, round_up):
 # A constant used as object metadata to indicate the object is raw bytes.
 OBJECT_METADATA_TYPE_RAW = b"RAW"
 
+# A constant used as object metadata to indicate the object is an actor handle.
+# TODO(fyrestone): Serialize the ActorHandle viathe custom type feature
```
Comment (raulchen) [inline python/ray/ray_constants.py:196]:
```suggestion
# TODO(fyrestone): Serialize the ActorHandle via the custom type feature
```
Can you also document that this needs to be in sync with the Java definition in the xxx.java file?

Comment (fyrestone) [reply python/ray/ray_constants.py]: The comments has been updated.

**Review by raulchen — APPROVED** [review]

**Pull request merged**